File: 03.AutomationTesting/07.Playwright-Project-Tests/Playwright-SauceDemo-Tests/tests/pages/InventoryPage.js
export class InventoryPage {
  constructor(page) {
    this.page = page;
    this.title = page.locator('[data-test="title"]');
    this.cartBadge = page.locator('[data-test="shopping-cart-badge"]');
    this.sortDropdown = page.locator('[data-test="product-sort-container"]');
    this.productNames = page.locator('[data-test="inventory-item-name"]');
    this.productPrices = page.locator('[data-test="inventory-item-price"]');
  }

  addToCartByTestId(productTestId) {
    return this.page.locator(`[data-test="add-to-cart-${productTestId}"]`).click();
  }

  // приема видимия label от dropdown-a, напр. 'Price (low to high)'
  sortBy(label) {
    return this.sortDropdown.selectOption({ label });
  }

  async getAllPrices() {
    const texts = await this.productPrices.allTextContents();
    return texts.map((t) => parseFloat(t.replace('$', '')));
  }

  getFirstProductName() {
    return this.productNames.first().textContent();
  }

  getFirstProductPrice() {
    return this.productPrices.first().textContent();
  }
}

File: 03.AutomationTesting/07.Playwright-Project-Tests/Playwright-CSharp-SauceDemo-Tests/Pages/InventoryPage.cs
using System.Globalization;
using Microsoft.Playwright;

namespace PlaywrightCSharpTests.Pages;

public class InventoryPage
{
    private readonly IPage _page;

    public ILocator Title { get; }
    public ILocator CartBadge { get; }
    public ILocator SortDropdown { get; }
    public ILocator ProductNames { get; }
    public ILocator ProductPrices { get; }

    public InventoryPage(IPage page)
    {
        _page = page;
        Title = page.Locator("[data-test='title']");
        CartBadge = page.Locator("[data-test='shopping-cart-badge']");
        SortDropdown = page.Locator("[data-test='product-sort-container']");
        ProductNames = page.Locator("[data-test='inventory-item-name']");
        ProductPrices = page.Locator("[data-test='inventory-item-price']");
    }

    public Task AddToCartAsync(string productTestId)
    {
        return _page.Locator($"[data-test='add-to-cart-{productTestId}']").ClickAsync();
    }

    // приема видимия label от dropdown-a, напр. "Price (low to high)"
    public Task SortByAsync(string label)
    {
        return SortDropdown.SelectOptionAsync(new SelectOptionValue { Label = label });
    }

    public async Task<string> GetFirstProductNameAsync()
    {
        return await ProductNames.First.TextContentAsync() ?? string.Empty;
    }

    public async Task<List<decimal>> GetAllPricesAsync()
    {
        var prices = await ProductPrices.AllTextContentsAsync();
        return prices
            .Select(p => decimal.Parse(p.Replace("$", ""), CultureInfo.InvariantCulture))
            .ToList();
    }
}
